mono/utils/mono-threads-coop.c

```diff
@@ -193,9 +223,16 @@ mono_threads_reset_blocking_start (void* stackdata)
 	if (!mono_threads_is_coop_enabled ())
 		return NULL;
 
+	info = mono_thread_info_current_unchecked ();
+
+#ifdef ENABLE_CHECKED_BUILD_GC
+	int reset_blocking_count = InterlockedIncrement (&coop_reset_blocking_count);
+	if (reset_blocking_count < 0)
+		reset_blocking_count = coop_reset_blocking_count = 1;
```

Comment:
So: I don't care if it's negative. I *do* care that it's nonzero. Wrapping around to negative is harmless, wrapping around to 0 is fatal because 0 is a special sentinel value that indicates this is a BlockingIgnore and we do not need to do any stack operations. I checked for < 0 as a way of checking for -1, I don't think I had any particular reason for this.

I did not worry about using interlocked on reset_blocking_count because collisions are basically harmless. The only downside to two threads using the same cookie is less helpful error reporting. Less helpful error reporting is bad enough to justify using interlocked in the main case, but it's not worth going to a bunch of bother to prevent an event where error reporting is potentially less helpful than it should be every 2^31 tries. (Also using interlockedexchange would not be sufficient to prevent a collision, if you consider what happens during a race.)

At a minimum, I think I should have commented why the check was present.